lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -628,250 +527,6 @@ assignMemoryLayouts(scf::ForOp &forOp, tt::CoarseSchedule &schedule,
   return loadToInfo;
```

Comment:
Unfortunately github won't let me comment in any place in the file, just around the changed code...
Comment to the line:
`// If that still didn't work, bail on pipelining this load.`
you should be able to assert right there I think, instead in separate loop checking if all loads from `loadsToPipeline` are in `loadToInfo`?

Reply:
In general, the logic in this loop seems off, we iterate over all the users, but if for the first user we are not able to find the encoding, we bail... Maybe we should decide upfront that we will only pipeline loads with single users, and simplify logic here?

Reply:
Yeah, the logic goes through all users, if there is one user we need to bail, it will continue to the next user. That is why outside of the loop we try to make sure all pipelined loads have loadInfo:
```
  // Make sure all loads in loadsToPipeline are in loadToInfo.
  for (auto *load : loadsToPipeline)
    assert(loadToInfo.count(load) &&
           "pipelined loads should have sharedEncoding");
```

Reply:
"only pipeline loads with single users" may be too restrictive?

Reply:
It may, maybe we should check if all the uses have the same stage instead. Again, probably ok to do as a follow-up